src/large_scale_condensation.jl

```diff
@@ -1,46 +1,99 @@
+    @unpack temp_tend, humid_tend = column  # tendencies to write into
 
     # 1. Tendencies of humidity and temperature due to large-scale condensation
-    for k = k_lsc:nlev  # Used to be 2:nlev in original Speedy
-        σₖ = σ_levels_full[k]
-        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ^2 - 1)  # Relative humidity threshold for condensation (Formula 24)
+    # skip the planetary boundary layer, k_lsc is lowest level for precip 
+    for k in eachlayer(column)[k_lsc:end]
+        
+        # Relative humidity threshold for condensation (Formula 24)
+        σₖ² = σ_levels_full[k]^2
+        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ² - 1)
         if k == nlev
             RH_threshold = max(RH_threshold, RH_thresh_boundary)
         end
 
         # Impose a maximum heating rate to avoid grid-point storm instability
         # This formula does not appear in the original Speedy documentation
-        humid_tend_max = 10σₖ^2 / 3600humid_relax_time
-
-        for j = 1:nlat, i = 1:nlon
-            humid_threshold = RH_threshold * sat_spec_humidity[i, j, k]  # Specific humidity threshold for condensation
-            if humid_grid[i, j, k] > humid_threshold
-                humid_tend_lsc[i, j, k] = -(humid_grid[i, j, k] - humid_threshold) / humid_relax_time                 # Formula 22
-                temp_tend_lsc[i, j, k] = -alhc / cp * min(humid_tend_lsc[i, j, k], humid_tend_max * pres_grid[i, j])  # Formula 23
-                cloud_top[i, j] = min(cloud_top[i, j], k)                                                             # Page 7 (last sentence)
-            else
-                humid_tend_lsc[i, j, k] = zero(NF)
-                temp_tend_lsc[i, j, k] = zero(NF)
-            end
+        humid_tend_max = 10σₖ² / 3600humid_relax_time
+        humid_threshold = RH_threshold * sat_spec_humidity[k]  # Specific humidity threshold for condensation
+
+        if humid[k] > humid_threshold
+            # accumulate in tendencies (nothing is added if humidity not above threshold)
+            humid_tend[k] += -(humid[k] - humid_threshold) / humid_relax_time       # Formula 22
+            temp_tend[k] += -alhc / cp * min(humid_tend[k], humid_tend_max*pres)    # Formula 23
+            column.cloud_top = min(column.cloud_top, k)                             # Page 7 (last sentence)
         end
     end
 
     # 2. Precipitation due to large-scale condensation
-    fill!(precip_large_scale, zero(NF))
-    for k = k_lsc:nlev
-        Δpₖ = pres_grid * σ_levels_thick[k]  # Formula 4
-        for j = 1:nlat, i = 1:nlon
-            precip_large_scale[i, j] += -1 / gravity * Δpₖ[i, j] * humid_tend_lsc[i, j, k]  # Formula 25
-        end
+    column.precip_large_scale = 0
+    for k in eachlayer(column)[k_lsc:end]
+        Δpₖ = pres*σ_levels_thick[k]                                        # Formula 4
+        column.precip_large_scale += -1 / gravity * Δpₖ * humid_tend[k]     # Formula 25
+    end
+end
```

Comment:
Do I understand this correctly that `precip_large_scale` is only computed here for output purposes but not actually used further? E.g. it's not written into any tendencies

Reply:
We should add a `precip_large_scale` field to DiagnosticVariables if we want to output that field.

Reply:
I'll add two fields to DiagnosticVariables.surface, `precip_large_scale` and `precip_convection` and then we can see whether we want to output them?

Reply:
Done with later commits 👇🏼